feat: Add QueryDSL config and Skeleton

jhkang1517 opened the pull request:
Signed-off-by: 강정호 <jhkang@gangjeonghoui-MacBookPro.local>

Commit:
:wrench: Add QueryDSL Configuration on build.gradle

Signed-off-by: 강정호 <jhkang@gangjeonghoui-MacBookPro.local>

Commit:
:wrench: Add querydsl Skeleton code

Signed-off-by: jhkang <jhkang@ieyes.co.kr>

Commit:
:ambulance: HotFix querydsl Skeleton code

Signed-off-by: jhkang <jhkang@ieyes.co.kr>

choipureum commented on src/main/java/com/makefire/anonymous/domain/user/repository/UserRepositoryImpl.java:
```diff
@@ -0,0 +1,27 @@
+    QUser user = QUser.user;
+
+    @Override
+    public List<User> getUserList() {
```
CURD 하나씩 해주시면 좋을것같습니당

choipureum replied:
오잉 insert될거같운데

jhkang1517 replied:
ㄹㅇ 되네요 ^^7 🙌🙌

choipureum commented on src/test/java/com/makefire/anonymous/support/RepositoryTestSupport.java:
```diff
@@ -18,5 +20,6 @@
 @DataJpaTest
 @AutoConfigureTestDatabase(replace = AutoConfigureTestDatabase.Replace.NONE)
 @ActiveProfiles("test")
+@Import(JpaTestConfig.class)
```
테스트도 도전!

jhkang1517 replied:
C 제외 URD 완료했습니당 👻👻

Commit:
:sparkle: feat: New Repository Method 'getUser'

Signed-off-by: jhkang <jhkang@ieyes.co.kr>

Commit:
:fire: Remove findById Test

Signed-off-by: jhkang <jhkang@ieyes.co.kr>

jhkang1517 committed:
:white_check_mark: feature: Add UserRepository 'update test'

Signed-off-by: jhkang <jhkang1517@gmail.com>

jhkang1517 committed:
:white_check_mark: feature: Add UserRepository 'delete test'

Signed-off-by: jhkang <jhkang1517@gmail.com>

Commit:
:white_check_mark: feat: New Repository Method 'insertUser' and test

Signed-off-by: jhkang <jhkang@ieyes.co.kr>

Commit:
:recycle: feat: Refactoring Querydsl codes

1. JpaConfiguration , QuerydslConfiguration 분리

2. Repository Test @Nested 적용

3. 불필요하게 선언되어 있는 package 삭제

4. build.gradle 가독성 향상

Signed-off-by: jhkang <jhkang@ieyes.co.kr>

choipureum commented on src/main/java/com/makefire/anonymous/domain/user/repository/UserRepositoryImpl.java:
```diff
@@ -18,6 +18,14 @@ public UserRepositoryImpl(JPAQueryFactory queryFactory) {
 
     QUser qUser = QUser.user;
```
Q클래스 인스턴스는 엄연히 기존 user 인스턴스와는 성질이 다릅니다. 따라서 이를 구분짓는 컨벤션은 적절한 것 같습니다.

choipureum commented:
LGTM 😉
